File: src/medium/t3/solve.go
package main

import "fmt"

func main() {
	var nums = []int{2, 7, 11, 15}
	var target = 9
	result := twoSum(nums, target)
	fmt.Println(result)
}

func twoSum(numbers []int, target int) []int {
	var i, j = 0, len(numbers) - 1
	for i < j {
		if sum := numbers[i] + numbers[j]; sum == target {
			break
		} else if sum > target {
			j--
		} else {
			i++
		}
	}
	return []int{i + 1, j + 1}
}

File: src/medium/t6/solve.py
from typing import List


class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        start = 0
        size = 2**31-1
        sum = 0
        for end in range(start, len(nums)):
            sum += nums[end]
            while sum >= target:
                size = min(size, end-start+1)
                sum -= nums[start]
                start += 1

        if size == 2**31-1:
            size = 0
        return size
